viewer/src/datamodels/cad/rendering/createSimpleGeometryMesh.ts

```diff
@@ -0,0 +1,105 @@
+/*!
+ * Copyright 2021 Cognite AS
+ */
+import * as THREE from 'three';
+
+import { ParsePrimitiveAttribute } from '@cognite/reveal-parser-worker';
+
+import { disposeAttributeArrayOnUpload } from '../../../utilities/disposeAttributeArrayOnUpload';
+import { filterPrimitivesOutsideClipBoxByBaseBoundsAndInstanceMatrix } from './filterPrimitives';
+import { Materials } from './materials';
+
+const quadVertexData = new Float32Array([
+  /* eslint-disable prettier/prettier */
+  -0.5, -0.5, 0.0,
+  0.5, -0.5, 0.0,
+  0.5, 0.5, 0.0,
+
+  0.5, 0.5, 0.0,
+  -0.5, 0.5, 0.0,
+  -0.5, -0.5, 0.0,
+  /*  eslint-enable prettier/prettier  */
+]);
+
+const quadVertexBufferAttribute = new THREE.Float32BufferAttribute(quadVertexData.buffer, 3);
+const baseBoundingBox = new THREE.Box3().setFromArray(quadVertexData);
+const primitiveAttributes: Map<string, ParsePrimitiveAttribute> = new Map([
+  ['color', { offset: 0, size: 4 * 3 }],
+  ['treeIndex', { offset: 12, size: 4 }],
+  ['normal', { offset: 16, size: 4 * 3 }],
+  ['instanceMatrix', { offset: 7 * 4, size: 16 * 4 }]
+]);
+
+export function createSimpleGeometryMesh(
+  attributeValues: Float32Array,
+  materials: Materials,
+  sectorBounds: THREE.Box3,
+  geometryClipBox: THREE.Box3 | null
+): THREE.Mesh | undefined {
+  const attributeByteValues = new Uint8Array(attributeValues.buffer);
+  const filteredByteValues = filterPrimitivesOutsideClipBoxByBaseBoundsAndInstanceMatrix(
+    attributeByteValues,
+    primitiveAttributes,
+    baseBoundingBox,
+    geometryClipBox
+  );
+  if (filteredByteValues.byteLength === 0) {
+    // No data, just skip
+    return undefined;
```

Comment:
Does it make sense to return an empty mesh here instead of undefined?

Reply:
By using `undefined` it's never added to the scene so there is no traversal before rendering (e.g. for pre-render sorting in ThreeJS). Probably no noticeable performance gain, but feels like the right thing to do for me.